from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtGui import QFont
from PyQt5.QtWidgets import QHeaderView, QAbstractItemView


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1172, 966)
        self.centralWidget = QtWidgets.QWidget(MainWindow)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.centralWidget.sizePolicy().hasHeightForWidth())
        self.centralWidget.setSizePolicy(sizePolicy)
        self.centralWidget.setObjectName("centralWidget")
        self.gridLayout = QtWidgets.QGridLayout(self.centralWidget)
        self.gridLayout.setContentsMargins(11, 11, 11, 11)
        self.gridLayout.setSpacing(6)
        self.gridLayout.setObjectName("gridLayout")
        self.videoQGroupBox = QtWidgets.QGroupBox(self.centralWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.videoQGroupBox.sizePolicy().hasHeightForWidth())
        self.videoQGroupBox.setSizePolicy(sizePolicy)
        self.videoQGroupBox.setObjectName("videoQGroupBox")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.videoQGroupBox)
        self.verticalLayout.setContentsMargins(11, 11, 11, 11)
        self.verticalLayout.setSpacing(6)
        self.verticalLayout.setObjectName("verticalLayout")
        # 修改video组件类型
        self.showVideo = QtWidgets.QLabel(self.videoQGroupBox)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(1)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.showVideo.sizePolicy().hasHeightForWidth())
        self.showVideo.setSizePolicy(sizePolicy)
        self.showVideo.setMinimumSize(QtCore.QSize(150, 0))
        self.showVideo.setObjectName("showVideo")
        self.verticalLayout.addWidget(self.showVideo)
        self.gridLayout.addWidget(self.videoQGroupBox, 0, 0, 3, 1)
        self.tableQGroupBox = QtWidgets.QGroupBox(self.centralWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.tableQGroupBox.sizePolicy().hasHeightForWidth())
        self.tableQGroupBox.setSizePolicy(sizePolicy)
        self.tableQGroupBox.setObjectName("tableQGroupBox")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.tableQGroupBox)
        self.verticalLayout_2.setContentsMargins(11, 11, 11, 11)
        self.verticalLayout_2.setSpacing(6)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.tableWidget = QtWidgets.QTableWidget(self.tableQGroupBox)
        # 增加表格结构
        # 使表格每一列扩展，使表头扩展
        self.tableWidget.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
        self.tableWidget.verticalHeader().setSectionResizeMode(QHeaderView.Stretch)
        # 设置表头字体和大小
        self.tableWidget.horizontalHeader().setFont(QFont("song", 12))
        font = self.tableWidget.horizontalHeader().font()
        font.setBold(True)
        self.tableWidget.horizontalHeader().setFont(font)
        # 设置表头背景色
        self.tableWidget.horizontalHeader().setStyleSheet("QHeaderView::section{background:skyblue;color: black;}")
        # 表内数据无法编辑
        self.tableWidget.setEditTriggers(QAbstractItemView.NoEditTriggers)
        self.tableWidget.setObjectName("tableWidget")
        self.tableWidget.setColumnCount(4)
        self.tableWidget.setRowCount(13)

        # 合并单元格,将最后一列单元格合并
        # 显示检测结果
        # self.tableWidget.setSpan(0, 4, 10, 1)
        # # 显示当前中箱有多少个ng的盒子
        # self.tableWidget.setSpan(10, 4, 3, 1)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.tableWidget.setHorizontalHeaderItem(3, item)
        # item = QtWidgets.QTableWidgetItem()
        # self.tableWidget.setHorizontalHeaderItem(4, item)
        self.tableWidget.verticalHeader().setVisible(False)
        self.verticalLayout_2.addWidget(self.tableWidget)
        self.gridLayout.addWidget(self.tableQGroupBox, 0, 1, 1, 5)
        self.logQHBpxLayout = QtWidgets.QHBoxLayout()
        self.logQHBpxLayout.setSpacing(6)
        self.logQHBpxLayout.setObjectName("logQHBpxLayout")
        self.logLevel = QtWidgets.QLabel(self.centralWidget)
        self.logLevel.setObjectName("logLevel")
        self.logQHBpxLayout.addWidget(self.logLevel)
        self.logComboBox = QtWidgets.QComboBox(self.centralWidget)
        self.logComboBox.setObjectName("logComboBox")
        self.logComboBox.addItem("")
        self.logComboBox.addItem("")
        self.logComboBox.addItem("")
        self.logQHBpxLayout.addWidget(self.logComboBox)
        self.gridLayout.addLayout(self.logQHBpxLayout, 1, 1, 1, 1)
        self.pushButton = QtWidgets.QPushButton(self.centralWidget)
        self.pushButton.setObjectName("pushButton")
        self.gridLayout.addWidget(self.pushButton, 1, 2, 1, 1)
        # self.label = QtWidgets.QLabel(self.centralWidget)
        # self.label.setObjectName("label")
        # self.gridLayout.addWidget(self.label, 1, 3, 1, 1)
        # self.label_2 = QtWidgets.QLabel(self.centralWidget)
        # self.label_2.setObjectName("label_2")
        # self.gridLayout.addWidget(self.label_2, 1, 4, 1, 1)
        # self.pushButton_2 = QtWidgets.QPushButton(self.centralWidget)
        # self.pushButton_2.setObjectName("pushButton_2")
        # self.gridLayout.addWidget(self.pushButton_2, 1, 5, 1, 1)
        self.currentLog = QtWidgets.QGroupBox(self.centralWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.currentLog.sizePolicy().hasHeightForWidth())
        self.currentLog.setSizePolicy(sizePolicy)
        self.currentLog.setObjectName("currentLog")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.currentLog)
        self.gridLayout_2.setContentsMargins(11, 11, 11, 11)
        self.gridLayout_2.setSpacing(6)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.showLogResult = QtWidgets.QListWidget(self.currentLog)
        self.showLogResult.setObjectName("showLogResult")
        self.gridLayout_2.addWidget(self.showLogResult, 0, 0, 1, 1)
        self.gridLayout.addWidget(self.currentLog, 2, 1, 1, 5)
        self.gridLayout.setColumnStretch(0, 5)
        self.gridLayout.setColumnStretch(1, 3)
        self.gridLayout.setRowStretch(0, 2)
        self.gridLayout.setRowStretch(1, 1)
        self.gridLayout.setRowStretch(2, 1)
        MainWindow.setCentralWidget(self.centralWidget)
        self.menuBar = QtWidgets.QMenuBar(MainWindow)
        self.menuBar.setGeometry(QtCore.QRect(0, 0, 1172, 23))
        self.menuBar.setObjectName("menuBar")
        MainWindow.setMenuBar(self.menuBar)
        self.mainToolBar = QtWidgets.QToolBar(MainWindow)
        self.mainToolBar.setToolButtonStyle(QtCore.Qt.ToolButtonTextBesideIcon)
        self.mainToolBar.setObjectName("mainToolBar")
        MainWindow.addToolBar(QtCore.Qt.TopToolBarArea, self.mainToolBar)
        self.statusBar = QtWidgets.QStatusBar(MainWindow)
        self.statusBar.setObjectName("statusBar")
        MainWindow.setStatusBar(self.statusBar)
        self.actStartCamera = QtWidgets.QAction(MainWindow)
        self.actStartCamera.setEnabled(False)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("resource/img/video_chat.ico"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actStartCamera.setIcon(icon)
        self.actStartCamera.setObjectName("actStartCamera")
        self.actQuit = QtWidgets.QAction(MainWindow)
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("resource/img/exit_24.ico"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actQuit.setIcon(icon2)
        self.actQuit.setObjectName("actQuit")
        # 新增窗口置顶按钮
        self.actWindowTopOn = QtWidgets.QAction(MainWindow)
        icon3 = QtGui.QIcon()
        icon3.addPixmap(QtGui.QPixmap("resource/img/pin.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actWindowTopOn.setIcon(icon3)
        self.actWindowTopOn.setObjectName("actWindowTopOn")
        # 新增窗口置顶取消按钮
        self.actWindowTopOff = QtWidgets.QAction(MainWindow)
        icon4 = QtGui.QIcon()
        icon4.addPixmap(QtGui.QPixmap("resource/img/offpin.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actWindowTopOff.setIcon(icon4)
        self.actWindowTopOff.setObjectName("actWindowTopOff")

        self.mainToolBar.addAction(self.actStartCamera)
        self.mainToolBar.addSeparator()
        self.mainToolBar.addAction(self.actQuit)
        self.mainToolBar.addSeparator()
        self.mainToolBar.addAction(self.actWindowTopOn)
        self.mainToolBar.addSeparator()
        self.mainToolBar.addAction(self.actWindowTopOff)

        self.retranslateUi(MainWindow)
        self.actQuit.triggered.connect(MainWindow.close)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "结直肠病灶检测追踪系统"))
        self.videoQGroupBox.setTitle(_translate("MainWindow", "视频检测展示"))
        self.tableQGroupBox.setTitle(_translate("MainWindow", "检测结果"))
        item = self.tableWidget.horizontalHeaderItem(0)
        item.setText(_translate("MainWindow", "病灶类别"))
        item = self.tableWidget.horizontalHeaderItem(1)
        item.setText(_translate("MainWindow", "病灶累计id"))
        item = self.tableWidget.horizontalHeaderItem(2)
        item.setText(_translate("MainWindow", "检测时间"))
        item = self.tableWidget.horizontalHeaderItem(3)
        item.setText(_translate("MainWindow", "所在位置"))
        # item = self.tableWidget.horizontalHeaderItem(4)
        # item.setText(_translate("MainWindow", "当前结果"))
        self.logLevel.setText(_translate("MainWindow", "日志级别"))
        self.logComboBox.setItemText(0, _translate("MainWindow", "info"))
        self.logComboBox.setItemText(1, _translate("MainWindow", "warning"))
        self.logComboBox.setItemText(2, _translate("MainWindow", "error"))
        self.pushButton.setText(_translate("MainWindow", "查询"))
        # self.label.setText(_translate("MainWindow", "当前检测的中箱流程为："))
        # self.label_2.setText(_translate("MainWindow", "TextLabel"))
        # self.pushButton_2.setText(_translate("MainWindow", "查询详情"))
        self.currentLog.setTitle(_translate("MainWindow", "实时日志"))
        self.actStartCamera.setText(_translate("MainWindow", "开启摄像头"))
        self.actStartCamera.setToolTip(_translate("MainWindow", "开启摄像头"))
        self.actQuit.setText(_translate("MainWindow", "退出"))
        self.actQuit.setToolTip(_translate("MainWindow", "退出"))
        self.actWindowTopOn.setText(_translate("MainWindow", "窗口置顶"))
        self.actWindowTopOn.setToolTip(_translate("MainWindow", "窗口置顶"))
        self.actWindowTopOff.setText(_translate("MainWindow", "取消窗口置顶"))
        self.actWindowTopOff.setToolTip(_translate("MainWindow", "取消窗口置顶"))

